internal/datastore/proxy/caching.go

```diff
@@ -99,41 +106,58 @@ func (r *nsCachingReader) ReadNamespace(
 	}
 
 	loaded := loadedRaw.(*cacheEntry)
-	return loaded.def, loaded.updated, loaded.notFound
+
+	var def core.NamespaceDefinition
+	err := def.UnmarshalVT(loaded.marshalledNsDef)
```

Comment:
In the case where the namespace was not found, what will have been marshaled in for us to unmarshal here?

Reply:
A NamespaceDefinition with all zero values for the fields. While it's not necessarily the most memory efficient (as I suppose using `nil` would be), it's the easiest to reason about, which I think is the more important thing for this codepath.

Reply:
Okay, mind adding a comment?

Reply:
I instead added a test because it actually does do `nil` and an empty slice. Not sure why I thought it did the zero value.